mlflow/openai/utils.py

```diff
@@ -99,3 +101,51 @@ def request(*args, **kwargs):
             return original(*args, **kwargs)
 
     return _mock_request(new=request)
+
+
+class _OAITokenHolder:
+    def __init__(self, api_type):
+        self.api_type = api_type
```

Comment:
Is this attribute used?

Reply:
What if a user already set `openai.api_key` to an azure token?

```python
from azure.identity import DefaultAzureCredential
import openai

# Request credential
default_credential = DefaultAzureCredential()
token = default_credential.get_token("https://cognitiveservices.azure.com/.default")

# Setup parameters
openai.api_type = "azure_ad"
openai.api_key = token.token
openai.api_base = "https://example-endpoint.openai.azure.com/"
openai.api_version = "2023-05-15"

import mlflow

...
```

Reply:
This is a good point. What I can do is to avoid token refresh if the token is configured manually like you just did. I think that's kind of an expected behavior, right? What do you think? Otherwise, we would attempt to refresh that token even though we don't know how the user got it in the first place.

Reply:
Agree！